common/src/main/java/com/sequenceiq/cloudbreak/certificate/PkiUtil.java

```diff
@@ -149,11 +156,12 @@ public static X509Certificate cert(KeyPair identity, String publicAddress, KeyPa
         }
     }
 
-    public static PKCS10CertificationRequest csr(KeyPair identity, String publicAddress) {
+    public static PKCS10CertificationRequest csr(KeyPair identity, String externalAddress) {
         try {
-            String name = String.format("C=US, CN=%s, O=Cloudera", publicAddress);
-            LOGGER.info("Generate CSR with X.500 distinguished name: '{}'", name);
-            return generateCsrWithName(identity, name);
+            String name = String.format("C=US, CN=%s, O=Cloudera", externalAddress);
+            List<String> sanList = List.of();
+            LOGGER.info("Generate CSR with X.500 distinguished name: '{}' and list of SAN: '{}'", name, String.join(",", sanList));
+            return generateCsrWithName(identity, name, sanList);
```

Comment:
I would accept some null check for the identity arg, since after passing it to the generateCsrWithName method that will use it directly. Possible NPE if for some reason the KeyPair is null (even if "it can't be null")

Reply:
My philosophy is that the public-facing methods should validate for null since if someone changes the code in the upper layers and passes null instead of some value, that could break the current code in runtime with unnecessary NPEs.
Since it's "just" a philosophy, I didn't mark these messages as a change request just giving my opinion. It's your decision to make such a change.